### PR TITLE
Backend improvements

### PR DESCRIPTION
**Pull request by jonathanjma**
- Fix unreads endpoint returning entries for the current user
- Add endpoint to get unreads from a certain group
- Move user groups and group invite accept/reject endpoints from user to groups (since it makes more sense to have them there)
- Update test suite and old queries

### CODE REVIEW

File: happiness-backend/api/dao/happiness_dao.py [hunk 1]
```diff
@@ -59,19 +59,41 @@ def get_happiness_by_date_range(user_ids: list[int], start: datetime, end: datet
...
     """
-    Returns a list of n Happiness objects (sorted from newest to oldest) given a list of User IDs.
+    Returns a paginated list of Happiness objects (sorted from newest to oldest) given a list of User IDs.
     Page variable can be changed to show the next n objects for pagination.
     """
     return list(db.paginate(
         select=(
-            select(Happiness).where(Happiness.user_id.in_(user_ids)).order_by(Happiness.timestamp.desc())
+            select(Happiness).where(Happiness.user_id.in_(user_ids))
+            .order_by(Happiness.timestamp.desc(), Happiness.user_id.asc())
```
Comment (zachseidner1) [inline happiness-backend/api/dao/happiness_dao.py:68]: Just out of curiosity, why did you do `user_id` ascending sort here?

File: happiness-backend/tests/test_reads.py [hunk 2]
```diff
@@ -190,13 +181,35 @@ def test_get_empty_unread_happiness_3(init_client):
     assert len(happiness_list) == 0
 
 
+def test_group_unread(init_client):
```
Comment (zachseidner1) [inline happiness-backend/tests/test_reads.py:184]: Would be good to add to this test where you try reading some entries and then see if they still show up in the unread entries.

**Review by zachseidner1 — APPROVED** [review]
Awesome work, thank you for improving the backend code base while you work as well! Seriously, great stuff. I left comments on a couple of things but nothing major. Approved 💰

File: happiness-backend/api/dao/happiness_dao.py (hunk 1 above)
Comment (jonathanjma) [reply happiness-backend/api/dao/happiness_dao.py:68]: i think sqlalchemy was arbitrary breaking ties in the sorting, which caused some pages to have duplicate entries

**Pull request merged**